playback/src/player.rs

```diff
@@ -527,69 +527,131 @@ impl PlayerInternal {
+                .audio_key()
+                .request(track.id, file_id);
+            let encrypted_file = AudioFile::open(&self.session, file_id);
 
-        let file_id = match track.files.get(&format) {
-            Some(&file_id) => file_id,
-            None => {
-                warn!("Track \"{}\" is not available in format {:?}", track.name, format);
-                return None;
-            }
-        };
 
-        let key = self
-            .session
-            .audio_key()
-            .request(track.id, file_id);
-        let encrypted_file = AudioFile::open(&self.session, file_id);
+            let encrypted_file = encrypted_file.wait().unwrap();
+            let key = key.wait().unwrap();
+            let mut decrypted_file = AudioDecrypt::new(key, encrypted_file);
 
+            let normalisation_factor = match NormalisationData::parse_from_file(&mut decrypted_file) {
+                Ok(normalisation_data) => NormalisationData::get_factor(&self.config, normalisation_data),
+                Err(_) => {
+                    warn!("Unable to extract normalisation data, using default value.");
+                    1.0 as f32
+                }
+            };
 
-        let encrypted_file = encrypted_file.wait().unwrap();
-        let key = key.wait().unwrap();
-        let mut decrypted_file = AudioDecrypt::new(key, encrypted_file);
+            let audio_file = Subfile::new(decrypted_file, 0xa7);
 
-        let normalisation_factor = match NormalisationData::parse_from_file(&mut decrypted_file) {
-            Ok(normalisation_data) => NormalisationData::get_factor(&self.config, normalisation_data),
-            Err(_) => {
-                warn!("Unable to extract normalisation data, using default value.");
-                1.0 as f32
-            }
-        };
+            let mut decoder = VorbisDecoder::new(audio_file).unwrap();
 
-        let audio_file = Subfile::new(decrypted_file, 0xa7);
+            if position != 0 {
+                match decoder.seek(position) {
+                    Ok(_) => (),
+                    Err(err) => error!("Vorbis error: {:?}", err),
+                }
+            }
 
-        let mut decoder = VorbisDecoder::new(audio_file).unwrap();
+            info!("Track \"{}\" loaded", track.name);
 
-        if position != 0 {
-            match decoder.seek(position) {
-                Ok(_) => (),
-                Err(err) => error!("Vorbis error: {:?}", err),
-            }
+            Some((decoder, normalisation_factor))
         }
+        else
+        {
+            let track = Episode::get(&self.session, track_id).wait().unwrap();
+            info!(
+                "Loading episode \"{}\" with Spotify URI \"spotify:track:{}\"",
+                track.name,
+                track_id.to_base62()
+            );
+
+            // episodes have lower bitrates and some other codecsAsMut
+            // ignoring for now. fix this
+            let format = match self.config.bitrate {
```

Comment:
Perhaps we ought to generally supported a subset of available bitrates? I've managed to come up with this which works for me.

```rust
            let formats = match self.config.bitrate {
                Bitrate::Bitrate96 => [FileFormat::OGG_VORBIS_96, FileFormat::OGG_VORBIS_160, FileFormat::OGG_VORBIS_320],
                Bitrate::Bitrate160 => [FileFormat::OGG_VORBIS_160, FileFormat::OGG_VORBIS_96, FileFormat::OGG_VORBIS_320],
                Bitrate::Bitrate320 => [FileFormat::OGG_VORBIS_320, FileFormat::OGG_VORBIS_160, FileFormat::OGG_VORBIS_96],
            };

            let format = formats.iter().find(|format| track.files.contains_key(format));

            let file_id = match format {
                Some(format) => *track.files.get(format).unwrap(),
                None => {
                    warn!("Track \"{}\" has no supported format.", track.name);
                    return None;
                }
            };